src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -206,41 +208,66 @@ export const TradeInput = memo(() => {
   const { sellAssetAccountId, buyAssetAccountId, setSellAssetAccountId, setBuyAssetAccountId } =
     useAccountIds()
 
-  const userAddress = useMemo(() => {
+  const sellAddress = useMemo(() => {
     if (!sellAssetAccountId) return ''
 
     return fromAccountId(sellAssetAccountId).account
   }, [sellAssetAccountId])
 
-  const { data: _isSmartContractAddress, isLoading: isAddressByteCodeLoading } =
-    useIsSmartContractAddress(userAddress)
+  const useReceiveAddressArgs = useMemo(
+    () => ({
+      fetchUnchainedAddress: Boolean(wallet && isLedger(wallet)),
+    }),
+    [wallet],
+  )
+
+  const { manualReceiveAddress, walletReceiveAddress } = useReceiveAddress(useReceiveAddressArgs)
+  const receiveAddress = manualReceiveAddress ?? walletReceiveAddress
+
+  const { data: _isSmartContractSellAddress, isLoading: isSellAddressByteCodeLoading } =
+    useIsSmartContractAddress(sellAddress)
+
+  const { data: _isSmartContractReceiveAddress, isLoading: isReceiveAddressByteCodeLoading } =
+    useIsSmartContractAddress(receiveAddress ?? '')
 
   const disableSmartContractSwap = useMemo(() => {
     // Swappers other than THORChain shouldn't be affected by this limitation
     if (activeSwapperName !== SwapperName.Thorchain) return false
 
     // This is either a smart contract address, or the bytecode is still loading - disable confirm
-    if (_isSmartContractAddress !== false) return true
+    if (_isSmartContractSellAddress !== false) return true
+    if (
+      tradeQuoteStep?.source === THORCHAIN_LONGTAIL_STREAMING_SWAP_SOURCE &&
+      _isSmartContractReceiveAddress !== false
+    )
+      return true
```

Comment:
might also need to update
`src/state/apis/swapper/helpers/validateTradeQuote.ts` -> `disableSmartContractSwap`

Reply:
Nice catch, will tackle 🙏🏽 also noticed we've apparently broken the smart contract checks in https://github.com/shapeshift/web/pull/5971/files#diff-b6f4d6672824399e71740531e3b09efe3ad3fabeb3ad6c4ce0763969f9d39ab7R210, will author a hotfix in the meantime

Reply:
- Restacked on top of https://github.com/shapeshift/web/pull/6081
- Added receive side checks and additional comments in 639cd282376759d556fb5d3efd1273e665992868
- made this PR *relate* to #5805 instead of closing it while at it, since we cannot yet test this until L1 -> EVM long-tails are implemented in https://github.com/shapeshift/web/issues/5892